src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+                            {
+                                var rows = new List<IReadOnlyDictionary<string, string>>();
+                                using (SqlDataReader reader = await getChangesCommand.ExecuteReaderAsync(token))
+                                {
+                                    while (await reader.ReadAsync(token))
+                                    {
+                                        rows.Add(SqlBindingUtilities.BuildDictionaryFromSqlRow(reader));
+                                    }
+                                }
+
+                                this._rows = rows;
+                            }
+
+                            // If changes were found, acquire leases on them.
+                            if (this._rows.Count > 0)
+                            {
+                                using (SqlCommand acquireLeasesCommand = this.BuildAcquireLeasesCommand(connection, transaction))
+                                {
+                                    await acquireLeasesCommand.ExecuteNonQueryAsync(token);
+                                }
+                            }
+                            transaction.Commit();
+                        }
+                        catch (Exception ex)
+                        {
+                            this._logger.LogError("Commit Exception Type: {0}", ex.GetType());
+                            this._logger.LogError("  Message: {0}", ex.Message);
+                            try
+                            {
+                                transaction.Rollback();
+                            }
+                            catch (Exception ex2)
+                            {
+                                this._logger.LogError("Rollback Exception Type: {0}", ex2.GetType());
+                                this._logger.LogError("  Message: {0}", ex2.Message);
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                // If there's an exception in any part of the process, we want to clear all of our data in memory and
+                // retry checking for changes again.
+                this._rows = new List<IReadOnlyDictionary<string, string>>();
+                this._logger.LogWarning($"Failed to check {this._userTable.FullName} for new changes due to error: {e.Message}");
+            }
+        }
+
+        private async Task ProcessChangesAsync(CancellationToken token)
+        {
+            if (this._rows.Count > 0)
+            {
+                this._state = State.ProcessingChanges;
+                IReadOnlyList<SqlChange<T>> changes = null;
+
+                try
+                {
+                    // What should we do if this fails? It doesn't make sense to retry since it's not a connection based
```

Comment:
Make sure to document this behavior in the design doc

Reply:
Tracking with #283.